src/walletSdk/Anchor/Sep6.ts

```diff
@@ -53,4 +57,87 @@ export class Sep6 {
       throw new ServerRequestFailedError(e);
     }
   }
+
+  /**
+   * Deposits funds using the SEP-6 protocol. Next steps by
+   * the anchor are given in the response.
+   *
+   * @param {object} options - The options for the deposit.
+   * @param {string} options.authToken - The authentication token.
+   * @param {Sep6DepositParams} options.params - The parameters for the deposit request.
+   *
+   * @returns {Promise<Sep6DepositResponse>} Sep6 deposit response, containing next steps if needed
+   * to complete the deposit.
+   *
+   * @throws {Error} If an unexpected error occurs during the deposit operation.
+   */
+  async deposit({
+    authToken,
+    params,
+  }: {
+    authToken: string;
+    params: Sep6DepositParams;
+  }): Promise<Sep6DepositResponse> {
+    return this.flow({ type: "deposit", authToken, params });
+  }
+
+  /**
+   * Initiates a withdrawal using SEP-6.
+   *
+   * @param {object} options - The options for the withdrawal operation.
+   * @param {string} options.authToken - The authentication token.
+   * @param {Sep6WithdrawParams} options.params - The parameters for the withdrawal request.
+   *
+   * @returns {Promise<Sep6WithdrawResponse>} Sep6 withdraw response.
+   */
+  async withdraw({
+    authToken,
+    params,
+  }: {
+    authToken: string;
+    params: Sep6WithdrawParams;
+  }): Promise<Sep6WithdrawResponse> {
+    return this.flow({ type: "withdraw", authToken, params });
+  }
+
+  private async flow({
+    type,
+    authToken,
+    params,
+  }: {
+    type: "deposit" | "withdraw";
+    authToken: string;
+    params: Sep6DepositParams | Sep6WithdrawParams;
+  }) {
+    const { transferServer } = await this.anchor.sep1();
+
+    try {
+      const resp = await this.httpClient.get(
+        `${transferServer}/${type}?${queryString.stringify(params)}`,
+        {
+          headers: {
+            "Content-Type": "application/json",
+            Authorization: `Bearer ${authToken}`,
+          },
+        },
+      );
+      return {
+        type: Sep6ResponseType.SUCCESS,
+        data: resp.data,
+      };
+    } catch (e) {
+      if (e.response?.data?.type === "non_interactive_customer_info_needed") {
+        return {
+          type: Sep6ResponseType.MISSING_KYC,
```

Comment:
Do we need to explicitly provide type to all this responses?

Reply:
it makes it a bit clearer for the user to handle with typescript 

eg. w/ a type in the response:
```
  const resp = deposit(...)
  if (resp.type === Sep6ResponseType.MISSING_KYC) {
    console.log((resp.data as Types.Sep6MissingKYC).fields);
  }
```

otherwise have to do something like:
```
if ("fields" in resp) {
   console.log((resp.data as Types.Sep6MissingKYC).fields);
  }
```

I'm not tied to it, I can remove it. Then the response will look like this:
```
export interface Sep6DepositResponse {
  data: Sep6DepositSuccess | Sep6MissingKYC | Sep6Pending;
}
```

Reply:
after thinking about it more I think you're right removing the type field and using a Union type is best, and is the more standard typescript way of handling

https://github.com/stellar/typescript-wallet-sdk/pull/77/commits/91b6268809501cc6b710d7abf4c874f1801c27ba